docs/spec/consensus/checkvalidators.md

```diff
@@ -0,0 +1,224 @@
+# Checking Validator Set (CheckVS)
+
+As part of the light client, the CheckVS procedure has to check, whether given
+two headers (whose heigts differ by more than 1), the LightClient can trust  the
+newer header under the assumption it trusted the old one.
+
+This document contains some math formulas. To ease reading, the file
+  /tendermint/docs/spec/pdfs/checkvalidators.pdf
+displays them correctly.
+
+## Definitions
+
+  * header fields
+    - $height$
+    - $bfttime$: the chain time when the header (block) was generated
+    - $V$: validator set containing validators.
+    - $nextV$: next validators
+  * $tp$: trusting period   
+  * for a time $t$, the predicate $correct(v,t)$ is true if the validator $v$
+    follows the protocol until time $t$ (we will see about recovery later).
+    Similarly we define $faulty(v,t)$.
+  * For each header $h$ it has locally stored, the LightClient stores whether
+    it trusts $h$. We write $trust(h) = true$, if this is the case.
+  * Validator fields. We will write a validator as a tuple $(v,p)$ such that
+    + $v$ is the identifier (we assume identifiers are unique in each validator set)
+    + $p$ is its voting power
+
+## LightClient Trusting Spec
+
+### LightClient Invariant
+ For each LightClient $l$ and each header $h$:
+if $l$ has set $trust(h) = true$,
+  then validators that are correct until time $h.bfttime + tp$ have more than two thirds of the voting power in $h.V$. (Or/and $h.nextV$)
+
+  Formally,
+  \[\sum_{(v,p) \in h.V \wedge \\correct(v,h.bfttime + tp)} p > 2/3 \sum_{(v,p) \in h.V} p\]
+
+  Equivalently,
+  \[\sum_{(v,p) \in h.V \wedge\\ faulty(v,h.bfttime + tp)} p < 1/3 \sum_{(v,p) \in h.V} p\]
+
+**Question:** What should be the precise assumption here. Is the invariant on $h.V$ or on $h.NextV$ or both?
+
+*Assumption:* If a header is properly generated, then the above equations hold.
+
+### Liveness
+
+*Draft:* If a header $h$ as been properly generated by the blockchain (and its age is less than the trusting period), then a correct LightClient will eventually set $trust(h) = true$.
+
+
+
+
+## High Level Solution
+
+Upon initialization, the LightClient is given a header *inithead* it trusts by
+social consensus. It is assumed that *inithead* satisfies the LightClient
+Invariant.
+
+When a LightClients sees a new header it has to decide whether to trust the new
+header. Trust can be obtained by (possibly) the combination of two methods.
+
+1. **an uninterrupted sequence of proof.** If a block is appended to the chain, where the last block
+is trusted (and properly committed by the old validator set in the next block),
+and the new block contains a new validator set, the new block is trusted if the LightClient knows all headers in the prefix.
+Intuitively, a trusted validator set is assumed to not chose a new validator set
```

Comment:
OK, but it's also the case, critically, that headers contain the hash of the next validator set, so the light client can verify that the validator set of block `n` (which they trust) signed off on the validator set of block `n + 1`, which provides some level of assurance (in practice) on what has happened in the state machine, e.g. that the new validators have put up stake which could be slashed.